lib/resources/resource_test.go

```diff
@@ -0,0 +1,86 @@
+package resources
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/kosyfrances/rundeck-zabbix/lib/zabbix"
+)
+
+// Test that we can get Rundeck resource model document in Yaml format
+// with hosts' information from Zabbix
+func TestMakeResource(t *testing.T) {
+	// mock api call
+	expected := zabbix.Results{
+		{
+			HostID:      "10261",
+			Host:        "dummy-host",
+			Name:        "dummy-host",
+			Description: "",
+		},
+	}
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		res := struct {
+			zabbix.Results `json:"result"`
+		}{expected}
+		err := json.NewEncoder(w).Encode(res)
+		if err != nil {
+			t.Fatalf("unable to write response. error: %v", err)
+		}
+	}))
+	defer ts.Close()
+
+	// generate temp file
+	tmpfile, err := ioutil.TempFile("", "testfile.*.yml")
+	if err != nil {
+		t.Fatalf("unable to create temporary file. error: %v", err)
+	}
+
+	defer os.Remove(tmpfile.Name()) // clean up
+	defer tmpfile.Close()           // close file
+
+	a := zabbix.NewAPI(ts.URL, "fake_zabbix_key")
+
+	res, err := a.GetHostsInfo()
+	if err != nil {
+		t.Fatalf("cannot get hosts info.\n%v", err)
+	}
+
+	Make(res, tmpfile.Name())
+	if err != nil {
```

Comment:
Hmm ... err at this point still references L50 above. I think the error you want to check here is the one returned from the preceding L55